src/faebryk/exporters/pcb/kicad/pcb.py

```diff
@@ -283,22 +284,27 @@ def get_property_uuid(
 
         # Add new nets -----------------------------------------------------------------
         logger.debug(f"New nets: {nets_added}")
+        existing_net_numbers = {net.number for net in pcb.kicad_pcb.nets}
+
         for net_name in nets_added:
-            # nl_net = nl_nets[net_name]
-            if removed_net_numbers:
-                net_number = removed_net_numbers.pop()
-            else:
-                net_number = len(pcb.kicad_pcb.nets)
+            # Find the first unused net number
```

Comment:
We do this a lot. While there's not currently a function for it in `util`, there probably should be

Something like (written freehand)
```python
def yield_missing(existing: Container, candidates: Iterable | None = None):
    for c in candidates:
        if c not in existing:
            yield c
```

The you can:
```suggestion
        available_numbers = iter(yield_missing(existing_net_numbers))
        for net_name in nets_added:
            net_number = next(available_numbers)
            
            # .... continue
            pcb_net = C_kicad_pcb_file.C_kicad_pcb.C_net(
                name=net_name,
                number=net_number,
            )
```

Notes:
- Candidates is `None` by default to default to a `count()`. We can't do this in the function declaration, because it won't be unique to each call
- You use it outside the for loop because then it preserves the start position to continue to the next number
- Ideally you use a set as the `existing` thing, because it's faster, but we don't need to